test-result-summary-client/src/Settings/Settings.jsx

```diff
@@ -81,24 +84,45 @@ export default class Settings extends Component {
         };
     }
 
-    onDelete = async ( key ) => {
+    onDelete = async ( record ) => {
         const { data } = this.state;
-        const target = data.find( item => item.key === key );
+        const target = data.find( item => item.key === record.key );
+
         if ( target && target._id ) {
-            const response = await fetch( `/api/deleteBuildListById?_id=${target._id}`, {
-                method: 'get'
-            } );
-            await response.json();
+            this.setState({ loading: true });
+            await this.deleteBuilds(record);
+            this.setState({ loading: false });
         }
-        this.setState( { data: data.filter( item => item.key !== key ) } );
+
+        this.setState( { data: data.filter( item => item.key !== record.key ) } );
+    }
+
+    deleteBuilds = async ( record ) => {
+        // delete URL from monitor list
+        const response = await fetch( `/api/deleteBuildListById?_id=${record._id}`, {
+            method: 'get'
+        } );
+        await response.json();
+
+         // delete build data from database
+        const fetchDeleteBuildData = await fetch(`/api/deleteBuildData?buildUrl=${record.buildUrl}`, {
+            method: 'get'
+        });
+        const deleteResponse = await fetchDeleteBuildData.json();
+        if (fetchDeleteBuildData.status === 400) {
+            message.error(deleteResponse.message);
+        } else if (fetchDeleteBuildData.status === 200){
+            message.success("Success!");
+        } 
```

Comment:
Please just display the return from the API. Format the message may cause the confusion.